### PR TITLE
DDF-1222 changed the login filter so that it attaches a java.security.auth.Subject to the thread

### PR DESCRIPTION
**Pull request by stustison**
@pklinef @shaundmorris

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf-platform/53)

### CODE REVIEW

File: security/core/security-core-impl/src/main/java/ddf/security/assertion/impl/SecurityAssertionImpl.java [hunk 1]
```diff
@@ -270,6 +275,25 @@ public Principal getPrincipal() {
         return null;
     }
 
+    @Override
+    public Set<Principal> getPrincipals() {
+        Set<Principal> principals = new HashSet<>();
+        Principal primary = getPrincipal();
+        principals.add(primary);
+        principals.add(new RolePrincipal(primary.getName()));
+        for (AttributeStatement attributeStatement : getAttibuteStatements()) {
```
Comment (pklinef) [inline security/core/security-core-impl/src/main/java/ddf/security/assertion/impl/SecurityAssertionImpl.java]: Are these loops null safe?

**Comment by pklinef** [comment]
:+1:

**Comment by shaundmorris** [comment]
Built and tested at runtime to make sure the Subject was accurate for permissions.

**Comment by shaundmorris** [comment]
:+1:

File: security/core/security-core-impl/src/main/java/ddf/security/assertion/impl/SecurityAssertionImpl.java (hunk 1 above)
Comment (stustison) [inline security/core/security-core-impl/src/main/java/ddf/security/assertion/impl/SecurityAssertionImpl.java]: Yes, the attributes are built in the code above this when the token is parsed out. It only adds them in if it reads it out of the XML in the first place.

**Pull request merged**